File: foliage-rpc/src/main/java/io/foliage/netty/rpc/protocol/MessageRequest.java
package io.foliage.netty.rpc.protocol;

import lombok.Data;

import java.io.Serializable;

@Data
public class MessageRequest implements Serializable {

    private String messageId;
    private String className;
    private String methodName;
    private Class<?>[] typeParameters;
    private Object[] parametersVal;
}

File: foliage-rpc/src/main/kotlin/io/foliage/netty/rpc/protocol/MessageRequest.kt
package io.foliage.netty.rpc.protocol

import org.lavenderx.foliage.nettyrpc.utils.formatToString

open class MessageRequest(val messageId: String) : java.io.Serializable {

    val className: String? = null
    val methodName: String? = null
    val parameterTypes: Array<Class<*>>? = null
    val parameterValues: Array<Any>? = null

    override fun toString(): String {
        return formatToString(this)
    }
}